src/python/base/tasks.py

```diff
@@ -364,6 +365,14 @@ def add_task(command, argument, job_type, queue=None, wait_time=None):
   if wait_time is None:
     wait_time = random.randint(1, TASK_CREATION_WAIT_INTERVAL)
 
+  job = data_types.Job.query(data_types.Job.name == job_type).get()
+  if not job:
+    raise Error(f'Job {job_type} not found.')
+
+  if job.is_external():
+    external_tasks.add_external_task(command, argument, job)
```

Comment:
nit: it might be nice if we can pass job name here, and move all the checks like job exists, is external, etc in that function as bail out conditions.

Reply:
Done!

Reply:
i think that makes some conditions in is_external more confusing because we need to skip adding tasks for all external jobs, even if that command is unsupported.. Also the job exists check seems beneficial in general if we're adding tasks.

If we do this the bailout conditions will look like:

```python
def add_external_task(...):
  if not job.is_external():
    return False

  if command != 'progression':
    return True   # this is confusing but necessary
```